### PR TITLE
xorg-x11-server: Add patches for CVE-2023-6478, CVE-2023-6377

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-6478
- https://nvd.nist.gov/vuln/detail/CVE-2023-6377

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/xorg-x11-server/xorg-x11-server.spec [hunk 1]
```diff
@@ -50,10 +50,11 @@ Patch5:         0001-autobind-GPUs-to-the-screen.patch
 # Because the display-managers are not ready yet, do not upstream
 Patch6:         0001-Fedora-hack-Make-the-suid-root-wrapper-always-start-.patch
 # Backports from current stable "server-1.20-branch":
-# <empty>
+# <empty> 
 
 # Backports from "master" upstream:
-Patch7: CVE-2023-1594.patch
+Patch7:         CVE-2023-1594.patch
+Patch8:         0001-manually-format-CVE-2023-6478-CVE-2023-6377.patch
```
Comment (mfrw) [inline SPECS/xorg-x11-server/xorg-x11-server.spec:57]: Wondering if our tooling recognizes this format for fixing CVEs :)

**Review by mfrw — APPROVED** [review]
LGTM

File: SPECS/xorg-x11-server/xorg-x11-server.spec (hunk 1 above)
Comment (mfrw) [inline SPECS/xorg-x11-server/xorg-x11-server.spec:57]: I can see 3 patch files added but only one patch file included ..

**Review by mfrw — CHANGES_REQUESTED** [review]

**Comment by dallasd1** [comment]
This change has been addressed in https://github.com/microsoft/CBL-Mariner/pull/7130

**Pull request closed without merging**